test/unit/requester-core.test.js

```diff
@@ -270,6 +284,40 @@ describe('requester util', function () {
             expect(requesterCore.getRequestOptions(request, options).lookup).to.be.a('function');
         });
 
+        it('should override lookup function and resolve localhost correctly when dnsOpts.all is true', function () {
+            var request = new sdk.Request({
+                    url: 'http://localhost:8080/random/path'
```

Comment:
We shall even add a test case when dnsOpts.all is false - then it picks up a single address string

Reply:
Yes. Already added it as part of already existing `it` statement. Line No: 219 and 240